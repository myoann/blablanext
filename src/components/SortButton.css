.sortButton {
  border: none;
  font-size: 20px;
  color: #00aff5;
  background-color: transparent;
  cursor: pointer;
}

.sortButton:hover {
  color: #008ec2;
}

.sortButtonSelected {
  font-weight: bold;
}

.sortButtonSelected:hover {
  color: #008ec2;
}
