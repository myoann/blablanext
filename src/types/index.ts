export enum Direction {
  Asc = "asc",
  Desc = "desc",
}

export type LatLng = {
  lat: number;
  lng: number;
};

export type RawTrip = {
  distance_in_meters: number;
  duration_in_seconds: number;
  link: string;
  waypoints: {
    place: {
      city: string;
      address: string;
      latitude: number;
      longitude: number;
      country_code: string;
    };
    date_time: string;
  }[];
  price: {
    amount: string;
    currency: string;
  };
  vehicle?: {
    make?: string;
    model?: string;
  };
};

export type ParsedTrip = {
  isCheapest: boolean;
} & RawTrip;

export enum SortType {
  Departure = "departure",
  Price = "price",
  Fastest = "fastest",
}
