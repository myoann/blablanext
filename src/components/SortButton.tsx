import { Direction, SortType } from "@/types";

import "./SortButton.css";

type Props = {
  direction: Direction;
  isActive: boolean;
  onClick: (filterType: SortType) => void;
  type: SortType;
};

const displayedFilterName = (type: SortType) => {
  switch (type) {
    case SortType.Departure:
      return "Departure";
    case SortType.Price:
      return "Price";
    case SortType.Fastest:
      return "Fastest";
  }
};

const SortButton = ({ direction, isActive, onClick, type }: Props) => {
  let sortIndicator = "";
  if (isActive) {
    sortIndicator = direction === Direction.Asc ? "↑" : "↓";
  }

  return (
    <button
      onClick={() => onClick(type)}
      className={`sortButton ${isActive && "sortButtonSelected"}`}
    >
      {displayedFilterName(type)} {sortIndicator}
    </button>
  );
};

export default SortButton;
