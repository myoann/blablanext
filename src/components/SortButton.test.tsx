import { render, screen } from "@testing-library/react";

import { Direction, SortType } from "@/types";

import SortButton from "./SortButton";

const defaultProps = {
  direction: Direction.Asc,
  isActive: false,
  onClick: jest.fn(),
  type: SortType.Departure,
};

describe("SortButton", () => {
  it("matches the snapshot", () => {
    const { container } = render(<SortButton {...defaultProps} />);

    expect(container).toMatchSnapshot();
  });

  it("calls the onClick function when clicked", () => {
    render(<SortButton {...defaultProps} />);

    const button = screen.getByRole("button");
    button.click();

    expect(defaultProps.onClick).toHaveBeenCalled();
  });

  it("displays the filter name", () => {
    render(<SortButton {...defaultProps} />);

    const button = screen.getByRole("button");
    expect(button).toHaveTextContent("Departure");
  });

  it("displays the filter name in descending order", () => {
    render(
      <SortButton {...defaultProps} direction={Direction.Desc} isActive />,
    );

    const button = screen.getByRole("button");
    expect(button).toHaveTextContent("Departure â†“");
  });

  it("displays the filter name for price", () => {
    render(<SortButton {...defaultProps} type={SortType.Price} />);

    const button = screen.getByRole("button");
    expect(button).toHaveTextContent("Price");
  });

  it("displays the filter name for fastest", () => {
    render(<SortButton {...defaultProps} type={SortType.Fastest} />);

    const button = screen.getByRole("button");
    expect(button).toHaveTextContent("Fastest");
  });
});
