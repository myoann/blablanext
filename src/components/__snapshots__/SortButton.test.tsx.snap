// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortButton matches the snapshot 1`] = `
<div>
  <button
    class="sortButton false"
  >
    Departure
     
    â†‘
  </button>
</div>
`;
